src/runtime/storage.js

```diff
@@ -103,6 +105,78 @@ export class Storage {
       resolve();
     });
   }
+
+  /**
+   * Stores the current time to local storage, under the storageKey provided.
+   * Removes times older than a week in the process.
+   * @param {string} storageKey
+   */
+  storeEvent(storageKey) {
+    return this.get(storageKey, /* useLocalStorage */ true).then((value) => {
+      const dateValues = this.filterOldValues_(
+        this.storedValueToDateArray_(value)
+      );
+      dateValues.push(Date.now());
+      const valueToStore = this.arrayToStoredValue_(dateValues);
+      this.set(storageKey, valueToStore, /* useLocalStorage */ true);
+    });
+  }
+
+  /**
+   * Retrieves the current time to local storage, under the storageKey provided.
+   * Filters out timestamps older than a week.
+   * @param {string} storageKey
+   * @return {!Promise<!Array<number>>}
+   */
+  getEvent(storageKey) {
+    return this.get(storageKey, /* useLocalStorage */ true).then((value) => {
```

Comment:
This method could be a little more compact
```suggestion
    return this.get(storageKey, /* useLocalStorage */ true).then((value) => this.filterOldValues_(this.storedValueToDateArray_(value)));
```